docs/testing/integration-tests.md

```diff
@@ -0,0 +1,200 @@
+serde_json = { version = "1.0", features = ["arbitrary_precision"] }
+tokio = { version = "1.18.1", features = ["full"] }
+tracing = "0.1"
+tracing-subscriber = { version = "0.3.11", features = ["env-filter"] }
+workspaces = "0.3.1"
+pkg-config = "0.3.1"
+
+[[example]]
+name = "integration-tests"
+path = "src/tests.rs"
+```
+
+The `tests.rs` file above will contain the integration tests, which will preserve state after each run if they are run in one file as in this setup. These can be run with the following command from the same level as the test `Cargo.toml` file:
+
+    cargo run --example integration-tests
+
+:::note
+In case you wish for a new state for a test run, you can create an additional `.rs` file in `tests/src` which you can reference as an additional `[[example]]` in the `Cargo.toml` file. You would similarly run these tests with the following command:
+
+    cargo run --example <name of the example>
+:::
+
+## Comparing an Example
+
+### Unit Test
+
+Let's take a look at a very simple unit test and integration test that accomplish the same thing. Normally you wouldn't duplicate efforts like this (as integration tests are intended to be broader in scope), but it will be informative.
+
+One of the simple examples on the <a href="https://near.dev" target="_blank">NEAR Examples landing page</a> is the <a href="https://examples.near.org/rust-counter" target="_blank">Rust counter</a>. We'll be using snippets from this repository to demonstrate simulation tests.
+
+First, note this unit test that tests the functionality of the `increment` method:
+
+```rust reference
+https://github.com/near-examples/rust-counter/blob/dad0cfd918fcc4c25611307aa07ad377b97ea52b/contract/src/lib.rs#L128-L139
+```
+
+The test above sets up the testing context, instantiates the smart contract's struct called `Counter`, calls the `increment` method, and makes an assertion that a number field on the struct is now `1`.
+
+Let's look at how this might be written with workspaces tests. The snippet below is a bit longer as it demonstrates a couple of things worth noting.
+
+### Workspaces Test
+
+```rust reference
+https://github.com/near-examples/rust-counter/blob/6a7af5a32c630e0298c09c24eab87267746552b2/integration-tests/rs/src/tests.rs#L6-L58
+```
+
+In the test above, we initialize the local blockchain environment with the `sandbox` constructor. Then the compiled smart contract `.wasm` file for the Rust Counter example is dev-deployed (newly created account) to the environment. The `root` account is fetched from the local environment later which is used to create accounts. This environment persists state between tests in this file and offers a similar environment to how a NEAR network node interacts. Next we see calls to a view and change method (with the `view!` and `call!` macros) along with an assertion checking the expected value. Tests are then listed sequentially, starting with the `increment` case.
```

Comment:
> Next we see calls to a view and change method (with the `view!` and `call!` macros)

There are no view calls though in the reference file and those macros are gone now

Reply:
Have removed this section